concrete/src/Http/Middleware/ThumbnailMiddleware.php

```diff
@@ -71,9 +72,21 @@ public function process(Request $request, DelegateInterface $frame)
 
                 if ($database) {
                     try {
-                        $paths = $database->executeQuery('SELECT * FROM FileImageThumbnailPaths WHERE isBuilt=0 LIMIT 5');
+                        if ($responseStatusCode === 404) {
```

Comment:
This block of changes makes this function pretty unruly. Can you break this functionality into a few methods so that the cyclomatic complexity doesn't go through the roof?